File: typescript/src/types/billing.ts
export interface Balance {
  available: string;
  locked: string;
  pending: string;
  currency: string;
}

export interface Transaction {
  uid: string;
  type: string;
  amount: string;
  currency: string;
  description?: string;
  created_at: string;
  metadata?: Record<string, unknown>;
}

export interface TransactionPage {
  items: Transaction[];
  total: number;
  page: number;
  size: number;
  pages: number;
}

File: python/src/azex/types/billing.py
"""Types for the billing resource."""

from __future__ import annotations

from typing import Any, Dict, List, Optional

from pydantic import BaseModel


class Balance(BaseModel):
    available: str
    locked: str
    pending: str
    currency: str


class Transaction(BaseModel):
    uid: str
    type: str
    amount: str
    currency: str
    description: Optional[str] = None
    created_at: str
    metadata: Optional[Dict[str, Any]] = None
